src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -45,19 +45,173 @@ struct extent_to_write_t {
   laddr_t addr = L_ADDR_NULL;
   extent_len_t len;
   std::optional<bufferlist> to_write;
+  std::optional<paddr_t> paddr;
 
   extent_to_write_t() = default;
   extent_to_write_t(const extent_to_write_t &) = default;
   extent_to_write_t(extent_to_write_t &&) = default;
 
   extent_to_write_t(laddr_t addr, bufferlist to_write)
-    : addr(addr), len(to_write.length()), to_write(to_write) {}
+    : addr(addr), len(to_write.length()), to_write(to_write), paddr(std::nullopt) {}
 
   extent_to_write_t(laddr_t addr, extent_len_t len)
-    : addr(addr), len(len) {}
+    : addr(addr), len(len), to_write(std::nullopt), paddr(std::nullopt) {}
+
+  extent_to_write_t(laddr_t addr, paddr_t paddr, extent_len_t len)
+    : addr(addr), len(len), to_write(std::nullopt), paddr(paddr) {}
 };
 using extent_to_write_list_t = std::list<extent_to_write_t>;
 
+struct split_info_t {
```

Comment:
I think this should be renamed to overwrite_plan_t, move immediately above ObjectDataHandler::overwrite.